### PR TITLE
[SPARK-40326][BUILD] Upgrade `fasterxml.jackson.version` to 2.13.4

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
upgrade `com.fasterxml.jackson.dataformat:jackson-dataformat-yaml` and `fasterxml.jackson.databind.version` from 2.13.3 to 2.13.4

### Why are the changes needed?
[CVE-2022-25857](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25857)

[SNYK-JAVA-ORGYAML](https://security.snyk.io/vuln/SNYK-JAVA-ORGYAML-2806360)

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by bjornjorgensen** [comment]
@LuciferYang

**Review by srowen — APPROVED** [review]
Looks good for Spark mater / 3.3.x

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

File: pom.xml [hunk 1]
```diff
@@ -170,7 +170,7 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.13.3</fasterxml.jackson.version>
+    <fasterxml.jackson.version>2.13.4</fasterxml.jackson.version>
     <fasterxml.jackson.databind.version>2.13.3</fasterxml.jackson.databind.version>
```
Comment (wangyum) [inline pom.xml:174]: Could you also update `fasterxml.jackson.databind.version` to 2.13.4?

**Review by wangyum — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (LuciferYang) [reply pom.xml]: I forgot the reason for adding `fasterxml.jackson.databind.version`, but do we still need two version properties related to `jackson`?

Comment (LuciferYang) [reply pom.xml]: Agree with @srowen

Comment (LuciferYang) [reply pom.xml]: No, as @srowen said, we should keep it

**Comment by ashutoshcipher** [comment]
The CVE [CVE-2022-25857](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25857) seems to be related to package `org.yaml:snakeyaml`. Does `fasterxml.jackson` has dependency on `org.yaml:snakeyaml` and  upgrading to `2.13.4` mitigating it?

**Comment by bjornjorgensen** [comment]
@ashutoshcipher Yes, SNYK give me this pr https://github.com/bjornjorgensen/spark/pull/25 where they well upgrade com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:
2.13.3 -> 2.13.4 after I change that and run ./dev/test-dependencies.sh --replace-manifest 
org.yaml:snakeyaml is upgraded.

**Comment by srowen** [comment]
Merged to master/3.3

**Pull request closed without merging**